### PR TITLE
Separate position creation from swap

### PR DESCRIPTION
**Pull request by lubkoll**

### CODE REVIEW

File: smart-contracts/osmosis/contracts/cl-vault/src/vault/range.rs [hunk 1]
```diff
@@ -217,58 +259,18 @@ pub fn do_swap_deposit_merge(
...
-        POSITION.save(
-            deps.storage,
-            &Position {
-                position_id: position_id.expect("position id should be set if no swap is needed"),
-                join_time: position.join_time,
-                claim_after: position.claim_after,
-            },
-        )?;
-
-        SWAP_DEPOSIT_MERGE_STATE.remove(deps.storage);
```
Comment (LaurensKubat) [inline smart-contracts/osmosis/contracts/cl-vault/src/vault/range.rs:268]: We should still make sure to clean up our state item after we're done using it

**Review by LaurensKubat — CHANGES_REQUESTED** [review]
generally lgtm, but we should remove  SWAP_DEPOSIT_MERGE_STATE when we're done using it, last remove was deleted as part of the refactor. Also I think this removes our ability to just open a position. But that's probably for a different PR

File: smart-contracts/osmosis/contracts/cl-vault/src/vault/range.rs [hunk 2]
```diff
@@ -195,13 +203,47 @@ pub fn handle_initial_create_position_reply(
 
     let unused_pair_balances = get_unused_pair_balances(&deps, &env, &pool_config)?;
 
+    if unused_pair_balances
```
Comment (LaurensKubat) [inline smart-contracts/osmosis/contracts/cl-vault/src/vault/range.rs:206]: This if condition means that we still have to swap some amount of tokens, combined with the middleware, we always swap 100% of the swappable tokens and thus cannot do easy inventory management. I don't think we should fix that for this PR

File: smart-contracts/osmosis/contracts/cl-vault/src/vault/range.rs (hunk 1 above)
Comment (lubkoll) [reply smart-contracts/osmosis/contracts/cl-vault/src/vault/range.rs:268]: The cleanup happens in `handle_iteration_create_position_reply`. Here it was just deleted as it was written before without needing it.

**Comment by lubkoll** [comment]
> generally lgtm, but we should remove SWAP_DEPOSIT_MERGE_STATE when we're done using it, last remove was deleted as part of the refactor. Also I think this removes our ability to just open a position. But that's probably for a different PR

Shouldn't have changed anything in the logic, just a refactor to separate concerns

File: smart-contracts/osmosis/contracts/cl-vault/src/vault/range.rs (hunk 1 above)
Comment (LaurensKubat) [reply smart-contracts/osmosis/contracts/cl-vault/src/vault/range.rs:268]: Sounds good, misread then

File: smart-contracts/osmosis/contracts/cl-vault/src/vault/range.rs (hunk 2 above)
Comment (LaurensKubat) [reply smart-contracts/osmosis/contracts/cl-vault/src/vault/range.rs:206]: Alright, think I misunderstood then

**Review by LaurensKubat — APPROVED** [review]
lgtm

**Pull request merged**